File: 민영/민석/구현/18312-시각.py
N, K = input().split()
N = int(N)
a = 0

for i in range(N + 1):
    if i < 10:
        i = '0' + str(i)
    for j in range(60):
        if j < 10:
            j = '0' + str(j)
        for k in range(60):
            if k < 10:
                k = '0' + str(k)
            if K in str(i) + str(j) + str(k):
                a += 1
print(a)

File: 민재/구현/18312_시각.swift
//
//  시각.swift
//  algo
//
//  Created by 김민재 on 2022/05/15.
//

import Foundation

let input = readLine()!.split(separator: " ").map {Int($0)!}
let N = input[0]
let K = input[1]


func makeString(number: Int) -> String {
    var twoString: String = "\(number)"
    if number < 10 {
        twoString = "0\(number)"
    }
    
    return twoString
}


var answer = 0
for h in 0...N {
    let hour = makeString(number: h)
    for m in 0..<60 {
        let minute = makeString(number: m)
        for s in 0..<60 {
            let second = makeString(number: s)
            if (hour + minute + second).contains("\(K)") {
                answer += 1
            }
        }
    }
}

print(answer)


